.github/scripts/findUnusedStyles.ts

```diff
@@ -0,0 +1,462 @@
+            }
+        }
+    }
+
+    private isStyleKey(key: string): boolean {
+        // Skip certain patterns that are likely not style keys
+        if (STYLE_KEY_SKIP_PATTERNS.includes(key)) {
+            return false;
+        }
+
+        // Must be a valid identifier
+        if (!VALID_IDENTIFIER_PATTERN.test(key)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private loadAllFileContents(): void {
+        // Load all source files EXCEPT style definition files
+        const allFilesPaths = `src/**/${STYLE_FILE_EXTENSIONS}`;
+        const allFiles = globSync(allFilesPaths, {
+            cwd: this.rootDir,
+        });
+
+        // Filter out style definition files, BUT keep utils/generators/themes for usage checking
+        const sourceFiles = allFiles.filter((file) => {
+            // Keep utils/generators/themes to check for main style usage
+            if (EXCLUDED_STYLE_DIRECTORIES.some((dir) => file.includes(dir))) {
+                return true;
+            }
+            // Exclude other style definition files (like index.ts itself)
+            return !file.includes('src/styles/');
+        });
+
+        console.log(`Loading ${sourceFiles.length} source files (including utils/generators/themes for usage checking)...`);
+
+        for (const file of sourceFiles) {
+            const fullPath = path.join(this.rootDir, file);
+            try {
+                // Check if it's actually a file (not a directory)
+                const stat = fs.lstatSync(fullPath);
+                if (!stat.isFile()) {
+                    console.warn(`Skipping ${file}: not a file`);
+                    continue;
+                }
+
+                const fileContent = fs.readFileSync(fullPath, 'utf8');
+                this.fileContents.set(file, fileContent);
+            } catch (error) {
+                console.warn(`Warning: Could not read source file ${file}:`, error);
+            }
+        }
+    }
+
+    private findStyleUsagesComprehensive(): void {
+        const allFiles = Array.from(this.fileContents.entries());
+        console.log(`Analyzing usage patterns for ${this.styleDefinitions.size} style keys across ${allFiles.length} files...`);
+
+        let processedFiles = 0;
+        for (const [file, content] of allFiles) {
+            this.analyzeFileForStyleUsage(file, content);
+            processedFiles++;
+
+            // Early termination: if all styles are found, stop processing
+            if (this.styleDefinitions.size === 0) {
+                console.log(`  ✅ All styles found as used!`);
+                break;
+            }
+
+            // Show progress every 100 files
+            if (processedFiles % 100 === 0) {
+                console.log(`  ${this.styleDefinitions.size} styles remaining...`);
+            }
+        }
+
+        console.log(`  Completed analysis.`);
+    }
+
+    private analyzeFileForStyleUsage(file: string, content: string): void {
```

Comment:
Why did we switch to plain text processing instead of AST parsing for this?

Reply:
I think overall the script will be more robust and easier to follow if it uses AST parsing both for find styles and searching for usages. I'm not a fan of the hybrid approach. I think it kind of gives us the worst of both worlds where we have the complexity of AST parsing, plus the complexity and brittleness of plaintext/regex parsing.